aspnet/2-structured-data/Models/DbBookStore.cs

```diff
@@ -46,12 +46,13 @@ public void Delete(long id)
         // [START list]
         public BookList List(int pageSize, string nextPageToken)
         {
-            var pageOfBooks = (null == nextPageToken ?
-                (from book in _dbcontext.Books orderby book.Id select book) :
-                (from book in _dbcontext.Books
-                 where book.Id > Int32.Parse(nextPageToken)
-                 orderby book.Id
-                 select book)).Take(pageSize + 1);
+            IQueryable<Book> query = from book in _dbcontext.Books orderby book.Id select book;
+            if (nextPageToken != null)
+            {
+                long previousBookId = long.Parse(nextPageToken);
+                query = from book in query where book.Id > previousBookId select book;
+            }
+            var pageOfBooks = query.Take(pageSize + 1);

```

Comment:
Why pageSize + 1?  To know if we should display a more button or not.

On Fri, Feb 12, 2016 at 2:08 PM, remi Taylor <notifications@github.com>
wrote:

> In aspnet/2-structured-data/Models/DbBookStore.cs
> <https://github.com/GoogleCloudPlatform/getting-started-dotnet/pull/4#discussion_r52805148>
> :
>
> > @@ -46,12 +46,13 @@ public void Delete(long id)
> >          // [START list]
> >          public BookList List(int pageSize, string nextPageToken)
> >          {
> > -            var pageOfBooks = (null == nextPageToken ?
> > -                (from book in _dbcontext.Books orderby book.Id select book) :
> > -                (from book in _dbcontext.Books
> > -                 where book.Id > Int32.Parse(nextPageToken)
> > -                 orderby book.Id
> > -                 select book)).Take(pageSize + 1);
> > +            IQueryable<Book> query = from book in _dbcontext.Books orderby book.Id select book;
> > +            if (nextPageToken != null)
> > +            {
> > +                long previousBookId = long.Parse(nextPageToken);
> > +                query = from book in query where book.Id > previousBookId select book;
> > +            }
> > +            var pageOfBooks = query.Take(pageSize + 1);
>
> Why query for more books than the paging size?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/getting-started-dotnet/pull/4/files#r52805148>
> .
>